docs/devguide/run_time_configuration.rst

```diff
@@ -36,17 +25,49 @@ OpenDDS configuration is concerned with three main areas:
    Each pluggable transport can be configured separately.
    See :ref:`config-transport` for details.
 
-The configuration file for OpenDDS is a human-readable ini-style text file.
+**********************
+Configuration Approach
+**********************
+
+..
+    Sect<7.1>
+
+Most of OpenDDS is configured through a key-value store.
+Keys are strings that are converted to a canonical form before being stored by 1) converting camel case to snake case, 2) capitalizing all alphabetic characers, 3) replacing all non-alphanumeric characters with underscores, and 4) stripping leading, trailing, and duplicate underscores.
+For example, the key ``!SectionInstance__PROPERTY`` is canonicalized into ``SECTION_INSTANCE_PROPERTY`` before being stored.
+Values are stored as strings and converted to other types as necessary.
+
+A key has the following parts:
+
+#. **section** -- The section describes the area of functionality that is being configured.
+
+#. **instance** -- (optional) The instance names a particular collection of configuration values.
+   Configuration keys that do not have an instance imply a singleton or global.
+
+#. **property** -- The property names a variable relative to the section and instance.
```

Comment:
I think we said we'd propbably be fine leaving the key vs property terms alone for now, but the fact that a ConfigStore key is explicitly defined to contain a property seems like it could be confusing. Should I change it so it's always referred to as a *property* so *key* is a sort of protected term in configuration that's specific to the ConfigStore?

Reply:
I think that's the right way to go.  I'm happy to change whatever I can in this PR if you describe what needs to be done.  Is is just looking for "key" and changing to property as necessary?

Reply:
I think I can change it in my next set of changes since I don't think these changes don't have any instances of that.

Reply:
#4564